src/conn/conn_cache.c

```diff
@@ -83,63 +87,122 @@ __cache_config_local(WT_SESSION_IMPL *session, bool shared, const char *cfg[])
 
-    /*
-     * Don't allow the dirty target to be larger than the overall target.
-     */
-    if (cache->eviction_dirty_target > cache->eviction_target)
-        cache->eviction_dirty_target = cache->eviction_target;
-
-    /*
-     * Sanity check the checkpoint target: don't allow a value lower than the dirty target.
-     */
-    if (cache->eviction_checkpoint_target > 0 &&
-      cache->eviction_checkpoint_target < cache->eviction_dirty_target)
-        cache->eviction_checkpoint_target = cache->eviction_dirty_target;
-
     WT_RET(__wt_config_gets(session, cfg, "eviction_dirty_trigger", &cval));
     cache->eviction_dirty_trigger = (double)cval.val;
     WT_RET(__cache_config_abs_to_pct(
       session, &(cache->eviction_dirty_trigger), "eviction dirty trigger", shared));
 
-    /*
-     * Don't allow the dirty trigger to be larger than the overall trigger or we can get stuck with
-     * a cache full of dirty data.
-     */
-    if (cache->eviction_dirty_trigger > cache->eviction_trigger)
-        cache->eviction_dirty_trigger = cache->eviction_trigger;
-
-    /* Configure updates target / trigger */
     WT_RET(__wt_config_gets(session, cfg, "eviction_updates_target", &cval));
     cache->eviction_updates_target = (double)cval.val;
     WT_RET(__cache_config_abs_to_pct(
       session, &(cache->eviction_updates_target), "eviction updates target", shared));
-    if (cache->eviction_updates_target < DBL_EPSILON)
-        cache->eviction_updates_target = cache->eviction_dirty_target / 2;
-
-    /* Don't allow the updates target to be larger than the eviction target. */
-    if (cache->eviction_updates_target > cache->eviction_target)
-        cache->eviction_updates_target = cache->eviction_target;
 
     WT_RET(__wt_config_gets(session, cfg, "eviction_updates_trigger", &cval));
     cache->eviction_updates_trigger = (double)cval.val;
     WT_RET(__cache_config_abs_to_pct(
       session, &(cache->eviction_updates_trigger), "eviction updates trigger", shared));
-    if (cache->eviction_updates_trigger < DBL_EPSILON)
+
+    WT_RET(__wt_config_gets(session, cfg, "eviction_checkpoint_target", &cval));
+    cache->eviction_checkpoint_target = (double)cval.val;
+    WT_RET(__cache_config_abs_to_pct(
+      session, &(cache->eviction_checkpoint_target), "eviction checkpoint target", shared));
+
+    /* Check for invalid configurations and automatically fix them to suitable values. */
+    if (cache->eviction_dirty_target > cache->eviction_target) {
+        if(FLD_ISSET(conn->debug_flags, WT_CONN_DEBUG_CONFIGURATION))
+            __wt_verbose_warning(session, WT_VERB_CONFIGURATION,
+              "config eviction_dirty_target=%f cannot exceed eviction_target=%f. Setting "
+              "eviction_dirty_target to %f.",
+              cache->eviction_dirty_target, cache->eviction_target, cache->eviction_target);
+        cache->eviction_dirty_target = cache->eviction_target;
+    }
```

Comment:
I'm wondering if the creation of macro for outputting the debug might make these corrective statements clearer, so a macro like:
```
#define CONFIG_DEBUG(msg, ...) \
  check if debugging is enabled and if so output msg
```
This would transform the code to something like:

```suggestion
    if (cache->eviction_dirty_target > cache->eviction_target) {
        cache->eviction_dirty_target = cache->eviction_target;
        CONFIG_DEBUG("config eviction_dirty_target=%f cannot exceed eviction_target=%f. Setting "
              "eviction_dirty_target to %f.", cache->eviction_dirty_target, cache->eviction_target, cache->eviction_target);        
    }
```

Note I also moved the change to be close to the condition checked, as the log messages are detailed which is very helpful, they are visually "noisy" in the code. Keeping the evaluation and the modification close to gether helps to clarify the purpose when scanning the code.

Reply:
I think creating a macro for debug is a good idea. If it’s clear enough from a readability standpoint that the messages are only printed when debug mode is enabled then I’m happy to modify it. In terms of moving the modification, we can't move it before printing the message. Otherwise, the invalid config would be overwritten, and the message would be using the post-modified correct config. We can bypass this by storing the previous value but I think it might be a bit unnecessary in this case.